app/services/supplementary-billing/reissue-invoice.service.js

```diff
@@ -0,0 +1,267 @@
+'use strict'
+
+/**
+ * Handles the reissuing of a single invoice
+ * @module ReissueInvoiceService
+ */
+const { randomUUID } = require('crypto')
+
+const ChargingModuleReissueInvoiceService = require('../charging-module/reissue-invoice.service.js')
+const ChargingModuleViewInvoiceService = require('../charging-module/view-invoice.service.js')
+const GenerateBillingInvoiceLicenceService = require('./generate-billing-invoice-licence.service.js')
+const GenerateBillingInvoiceService = require('./generate-billing-invoice.service.js')
+
+/**
+ * Handles the reissuing of a single invoice
+ *
+ * This service raises a reissue request with the Charging Module for an invoice (referred to as the "source" invoice)
+ * which creates 2 invoices on the CM side: a "cancelling" invoice (ie. one which cancels out the source invoice by
+ * being a credit for the same amount instead of a debit, or vice-versa) and a "reissuing" invoice (ie. the replacement
+ * for the source invoice).
+ *
+ * This request is all that needs to be done on the CM, so we then move on to our side of things. We create two new
+ * billing invoices on our side (a cancelling invoice and a reissuing invoice), and for each one we re-create the
+ * billing invoice licences and transactions of the source invoice. While doing this we are loading up a `dataToReturn`
+ * object with the new invoices, invoice licences and transactions.
+ *
+ * Once the invoice has been handled, we mark the source invoice as `rebilled` in the db (note that "rebill" is legacy
+ * terminology for "reissue") along with the `originalBillingInvoiceId` field; if this is empty then we update it with
+ * the source invoice's billing invoice id, or if it's already filled in then we leave it as-is. This ensures that if an
+ * invoice is reissued, and that reissuing invoice is itself reissued, then `originalBillingInvoiceId` will still point
+ * directly to the original source invoice.
+ *
+ * @param {module:BillingInvoiceModel} sourceInvoice The invoice to be reissued
+ * @param {module:BillingBatchModel} originalBillingBatch The billing batch that the original invoice belongs to
+ * @param {module:BillingBatchModel} reissueBillingBatch The billing batch that the new invoices should belong to
+ *
+ * @returns {Object} dataToReturn Data that has been generated while reissuing the invoice
+ * @returns {Object[]} dataToReturn.billingInvoices Array of billing invoices
+ * @returns {Object[]} dataToReturn.billingInvoiceLicences Array of billing invoice licences
+ * @returns {Object[]} dataToReturn.transactions Array of transactions
+ */
+
+async function go (sourceInvoice, originalBillingBatch, reissueBillingBatch) {
```

Comment:
This is tied to my other comment about the `originalBillingBatch`.

When we send the request to the Charging Module we need to provide it with the bill run ID that we want the invoices reissued against. In this context that will be `reissueBillingBatch.externalId`.

The Charging Module will then create the invoices against it, which means we should use the same `reissueBillingBatch.externalId` in `_sendViewRequest()`.

So, again I think we can remove `originalBillingBatch` as it is redundant in this context.